.gitlab-ci.yml

```diff
@@ -6,329 +6,167 @@ stages:
+    SPACK_ENV_NAME: openmpi-fortran-gcc
 
-install_gnu_skylake_fort_kokkos_cxx17:
+openmpi_cuda_gcc_ampere:
+  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
   needs: [download_prereq]
-  extends: .test
   variables:
-    <<: *skylake
-    <<: *python
-    <<: *fort
-    SINGULARITY_EOS_CXX_STANDARD: 17
-    SINGULARITY_EOS_SPACK_SPEC: "singularity-eos@main+python+mpi+tests+kokkos+spiner+hdf5+kokkos-kernels%gcc@=${SINGULARITY_EOS_GCC_VERSION} ^kokkos std=${SINGULARITY_EOS_CXX_STANDARD} +openmp ^openmpi@${SINGULARITY_EOS_OPENMPI_VERSION}"
+    SPACK_ENV_NAME: openmpi-cuda-gcc-ampere
+    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p shared-gpu-ampere"
 
-install_gnu_skylake_fort_static_kokkos:
+openmpi_cuda_gcc_volta:
+  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
   needs: [download_prereq]
-  extends: .test
   variables:
-    <<: *skylake
-    <<: *python
-    <<: *fort
-    SINGULARITY_EOS_SPACK_SPEC: "singularity-eos@main+python+mpi+tests+kokkos+spiner+hdf5+kokkos-kernels%gcc@=${SINGULARITY_EOS_GCC_VERSION} ^kokkos std=${SINGULARITY_EOS_CXX_STANDARD} ~shared +openmp ^kokkos-kernels~shared ^openmpi@${SINGULARITY_EOS_OPENMPI_VERSION}"
-
-install_gnu_skylake_nofort:
+    SPACK_ENV_NAME: openmpi-cuda-gcc-volta
+    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p volta-x86 -C cpu_family:haswell"
+
+# currently disabled due to very long compilation times
+#openmpi_fortran_cuda_gcc_ampere:
+#  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
+#  needs: [download_prereq]
+#  variables:
+#    SPACK_ENV_NAME: openmpi-fortran-cuda-gcc-ampere
+#    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p shared-gpu-ampere"
+#
+#openmpi_fortran_cuda_gcc_volta:
+#  extends: [.ascgit_job, .darwin_job, .report_tests, .build_and_test]
+#  needs: [download_prereq]
+#  variables:
+#    SPACK_ENV_NAME: openmpi-fortran-cuda-gcc-volta
+#    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p volta-x86 -C cpu_family:haswell"
+
+# EXAMPLE of how to modify an existing Spack environment and/or set CMake options
+#openmpi_cuda_gcc_volta_noclosure:
+#  stage: build_n_test
+#  extends: [.ascgit_job, .darwin_job, .report_tests,]
+#  needs: [download_prereq]
+#  variables:
+#    SPACK_ENV_NAME: openmpi-cuda-gcc-volta
+#    SCHEDULER_PARAMETERS: "-N 1 --qos=debug -p volta-x86 -C cpu_family:haswell"
+#    SINGULARITY_EOS_SPEC: "%gcc@=12.2.0 +mpi+cuda+kokkos+spiner+hdf5+kokkos-kernels+tests~closure~fortran build_extra=sesame,stellarcollapse cuda_arch=70"
+#  script:
+#   - source .gitlab/build_and_test.sh --until env ${CLUSTER} ${SPACK_ENV_NAME}
+#   - spack config add "packages:singularity-eos:require:'${SINGULARITY_EOS_SPEC}'"
+#   - spack concretize -f --fresh
+#   - spack_cmake_configure
+#   - cmake_build -DSINGULARITY_BUILD_CLOSURE=off -DSINGULARITY_USE_FORTRAN=off
+#   - cmake_test
+#   - cmake_install
```

Comment:
Can you add a description of how this works to the sphinx docs? Big picture it would be nice to have something farily descriptive of how the CI works. But at the very least we should have a "what do you need to do for the CI" section in the contributing page.

Reply:
Added some basic docs.

Reply:
Looks good. :+1: